EELS/EELS_elemental_mapping.ipynb

```diff
@@ -0,0 +1,789 @@
+    "\n",
+    "For more information about the material system and how the results of this processing can be used, see the paper: https://doi.org/10.1016/j.cattod.2019.02.045\n",
+    "\n",
+    "This notebook requires `HyperSpy` 2.0 or higher. It also requires `eXSpy` which is a library which contains all of HyperSpy's EELS and EDX functionality.\n",
+    "\n",
+    "The main objective is to show how the functionality of HyperSpy can be used to create relative elemental maps from EEL spectra.\n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "jp-MarkdownHeadingCollapsed": true
+   },
+   "source": [
+    "### Changes\n",
+    "\n",
+    "* 2017/09/27: Initial version by Ida Hjorth\n",
+    "* 2019/11/14: Update to HyperSpy 1.5, and minor improvements to text by Magnus Nord\n",
+    "* 2024/3/16: Update to work with HyperSpy 2.0, by Magnus Nord"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Table of contents"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "1. <a href='#load'> Load data</a> \n",
+    "2. <a href='#fitting'>Fitting Cu and Zn Hartree-Slater cross sections</a>\n",
+    "3. <a href='#save_restore'>Saving and restoring the model</a>\n",
+    "4. <a href='#plotting'>Creating elemental maps from the model</a>"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# <a id='load'></a>1. Load data"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%matplotlib widget\n",
+    "import numpy as np\n",
+    "import hyperspy.api as hs"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "s_eels = hs.load('datasets/CuZn_EELS_mapping_tutorial.hspy')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Making elemental maps for this map is a bit tricky, due to the fairly weak edges, and the overlap between the Cu and Zn edges."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "s_eels.plot()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Generate synthetic zero loss peak for low-loss signal (if lacking)\n",
```

Comment:
Is it really necessary to create a synthetic zero loss peak here? Since it doesn't have a bulk plasmon, it will not change the shape of the edges - the specimen doesn't seem very thick anyway. It will smooth a bit the model, but that doesn't really matter for these edges since there are delayed edge and the cross section doesn't have any fine structure?

Add a comment that the specimen is not thick in this case and that the effect of plural scattering on the data is weak/negligible?

Reply:
I removed this part.

Previously we used the H-S cross sections here, which was quite a lot "sharper". And effectively smoothing out the model with a synthetic zero loss peak improve the fitting a bit. However, with the new GOSH, this is no longer an issue.